taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -955,12 +955,18 @@ void TaskCodeGenLLVM::visit(PrintStmt *stmt) {
     if (std::holds_alternative<Stmt *>(content)) {
       auto arg_stmt = std::get<Stmt *>(content);
       auto value = llvm_val[arg_stmt];
+      auto value_type = value->getType();
       if (arg_stmt->ret_type->is<TensorType>()) {
         auto dtype = arg_stmt->ret_type->cast<TensorType>();
         auto elem_type = dtype->get_element_type();
         for (int i = 0; i < dtype->get_num_elements(); ++i) {
-          auto elem_value = builder->CreateExtractElement(value, i);
-          args.push_back(value_for_printf(elem_value, elem_type));
+          if (llvm::dyn_cast<llvm::VectorType>(value_type)) {
```

Comment:
Let's use info in Taichi to determine the type. Similar case: https://github.com/taichi-dev/taichi/blob/bb1d101af0172e14f4eef3286ff4ce60117d51cd/taichi/runtime/llvm/llvm_context.cpp#L155-L158